mbed_devices/_internal/resolve_target.py

```diff
@@ -50,7 +43,10 @@ def _build_target_resolver(candidate: CandidateDevice) -> Callable:
         online_id = _extract_online_id(all_files_contents)
         return functools.partial(get_target_by_online_id, slug=online_id.device_slug, target_type=online_id.device_type)
     except OnlineIdNotFound:
-        raise UnableToBuildResolver
+        pass
+
+    # Product code might be the first 4 characters of the serial number
+    return functools.partial(get_target_by_product_code, candidate.serial_number[:4])
```

Comment:
I think this `functools.partial` approach is too complex and I should've flagged it last review. Why not just have resolve target call the right function instead of this `build_target_resolver` fancy stuff? Maybe I'm missing something...

Reply:
Oh, you're right. That'll actually work better. 👍